inc/saiuburst.h

```diff
@@ -0,0 +1,401 @@
+     *
+     * @type sai_object_id_t
+     * @flags MANDATORY_ON_CREATE | CREATE_AND_SET
+     * @objects SAI_OBJECT_TYPE_TAM
+     */
+    SAI_TAM_HISTOGRAM_ATTR_TAM_ID = SAI_TAM_HISTOGRAM_ATTR_START,
+
+    /**
+     * @brief Buffers/Statistics for inclusion in histogram
+     *
+     * Specifies the Statistics/Types for a histogram.
+     *
+     * @type sai_object_list_t
+     * @flags CREATE_AND_SET
+     * @objects SAI_OBJECT_TYPE_TAM_STAT
+     * @default empty
+     */
+    SAI_TAM_HISTOGRAM_ATTR_STAT_TYPE,
+
+    /**
+     * @brief Histogram Bins Lower Boundaries
+     *
+     * List of lower boundary of each bin for this HISTOGRAM in
+     * number referred object units. The upper boundary of a bin is
+     * the lower boundary of next bin. The upper boundary of the
+     * last bin is infinity.
+     *
+     * @type sai_u32_list_t
+     * @flags MANDATORY_ON_CREATE | CREATE_AND_SET
+     * @default empty
+     */
+    SAI_TAM_HISTOGRAM_ATTR_BIN_BOUNDARY,
+
+    /**
+     * @brief Histogram Resolution
+     *
+     * The resolution to read measure statistics for inclusion in histogram
+     * May be every statistic object (as packet or microburst) or time interval
+     * (as for buffer current usage level)
+     *
+     * @type sai_uint32_t
+     * @flags CREATE_AND_SET
+     * @default 0
+     */
+    SAI_TAM_HISTOGRAM_ATTR_RESOLUTION,
+
+    /**
+     * @brief Histogram Clear-On-Read Mode
+     *
+     * If true the histogram bin values are clear on read/transport.
+     * Otherwise values in bins are counting continuously.
+     *
+     * @type bool
+     * @flags CREATE_AND_SET
+     * @default true
+     */
+    SAI_TAM_HISTOGRAM_ATTR_CLEAR_MODE,
+
+    /**
+     * @brief Transporter Object
+     *
+     * Provides the transporter object for this histogram. When the
+     * data shot is made, this transporter will be used to 'copy'
+     * the data to the 'transporter-desired' location. In the
+     * absence of a transporter, the tracker's default transporter
+     * will be used (DEFAULT).
+     *
+     * @type sai_object_id_t
+     * @flags CREATE_AND_SET
+     * @objects SAI_OBJECT_TYPE_TAM_TRANSPORTER
+     * @allownull true
+     * @default SAI_NULL_OBJECT_ID
+     */
+    SAI_TAM_HISTOGRAM_ATTR_TRANSPORTER,
+
+    /**
+     * @brief End of Attributes
+     */
+    SAI_TAM_HISTOGRAM_ATTR_END,
+
+    /** Custom range base value */
+    SAI_TAM_HISTOGRAM_ATTR_CUSTOM_RANGE_START = 0x10000000,
+
+    /** End of custom range base */
+    SAI_TAM_HISTOGRAM_ATTR_CUSTOM_RANGE_END
+
+} sai_tam_histogram_attr_t;
+
+/**
+ * @brief Create and return a microburst object
+ *
+ * @param[out] tam_microburst_id Microburst object
+ * @param[in] attr_count Number of attributes
+ * @param[in] attr_list Array of sai_tam_microburst_attr_t attributes
+ *
+ * @return #SAI_STATUS_SUCCESS on success, failure status code on error
+ */
+typedef sai_status_t (*sai_create_tam_microburst_fn)(
+        _Out_ sai_object_id_t *tam_microburst_id,
```

Comment:
function signature is missing switch_id parameter, take a look at other "create" functions